master/master.go

```diff
@@ -126,3 +144,37 @@ func instanceIds(instances []*ec2.Instance) []*string {
 	}
 	return instanceIds
 }
+
+func (m *Master) getTaskToSchedule() int {
+	for idx, t := range m.newTasks {
+		if !m.jobs[t.JobID].reachedMaxTasks() {
+			return idx
+		}
+	}
+	return -1
+}
+
+func (m *Master) slotsAvailable() bool {
```

Comment:
Nit pick: you could combine this method with the loop in the scheduleTask method. Otherwise you iterate `m.instances` twice.